samples/oauth_rails_vethealth/app/models/oauth_callback.rb

```diff
@@ -0,0 +1,47 @@
+class OauthCallback < ApplicationRecord
+  serialize :response_body_raw
+
+  has_one :authentication
+
+  def auth(protect_secrets=true)
+    pass =
+      if protect_secrets
+        '<shhhhhhh>'
+      else
+        ENV['va_developer_client_secret']
+      end
+    { username: ENV['va_developer_client_id'], password: pass }
+  end
+
+  def post_body
+    @post_body ||= {
+      code: self.code,
+      state: self.state,
+      grant_type: 'authorization_code',
+      redirect_uri: 'http://localhost:3000/callback'
+    }
+  end
+
+  def fetch_access_token!(session)
+    oauth_post = HTTParty.post(self.oauth_url, { basic_auth: auth(false), body: post_body })
+    self.response_body_raw = oauth_post.to_h
+    self.response_code = oauth_post.code
+    save!
+
+    if oauth_post.ok?
+      attributes = Authentication.attributes_from_oauth(oauth_post)
+      self.build_authentication(attributes)
+      if self.authentication.validate_session(session)
```

Comment:
Does this code not run when you call `.save!` on the Authentication instance? Shouldn't it?

Reply:
I went back and forth mentally on this being a proper validation.  But since it's validating if the cookie is valid for the saved authentication instead of the session being a property of the Authentication, I decided it wasn't a validation in the rails-sense.  But I don't know a better word than "validate" to use here?

Reply:
Ahh, I see. Maybe just add some comments about it.